frontend/dockerfile/parser/line_parsers.go

```diff
@@ -282,7 +282,7 @@ func parseJSON(rest string) (*Node, map[string]bool, error) {
 	}
 
 	var myJSON []interface{}
-	if err := json.NewDecoder(strings.NewReader(rest)).Decode(&myJSON); err != nil {
+	if err := json.Unmarshal([]byte(rest), &myJSON); err != nil {
```

Comment:
Contrary to https://github.com/moby/moby/pull/12793#discussion_r29163314 , `json.Unmarshal` is actually faster and allocates significantly less memory than `NewDecoder()`(always allocates at least 1KB) when parsing small JSON values.

Reply:
Perhaps worth updating some code in places then 😅